CHANGES.md

```diff
@@ -16,6 +16,8 @@
 - Infer release versions are inferred from VCS tags. This change allows using
   `dune-release` on projects that do not use the changelog or have it in a
   different format.  (#381, #383 @Leonidas-from-XIV)
+- Use tag for creating a GitHub release instead of the project version (#387,
```

Comment:
I've looked at this and you're right. The case where this would trigger was extremely rare (therefore, no wonder this hasn't been reported before. But also nice that adding a distinction between versions and tags has made this apparent). You're totally right that my changelog entry was misleading and actually wrong. I've rewritten the entry to be more clear and highlighting the user-visible change better.